src/wrapper/jnienv.rs

```diff
@@ -949,8 +949,10 @@ impl<'a> JNIEnv<'a> {
         Ok(ptr)
     }
 
-    #[allow(clippy::not_unsafe_ptr_arg_deref)]
     /// Unpin the array returned by `get_string_utf_chars`.
+    ///
+    /// It is safe to dereference a pointer that comes from `get_string_utf_chars`.
```

Comment:
The documentation says that, therefore I'd add it to the annotation itself as an impl comment instead.